app/uk/gov/hmrc/residencenilratebandcalculator/controllers/IHT435Controller.scala

```diff
@@ -0,0 +1,151 @@
+      "IHT435_10_03",
+      "IHT435_10_04",
+      "IHT435_10_05",
+      "IHT435_10_06",
+      "IHT435_10_07"
+    ),
+    Constants.exemptionsAndReliefClaimedId -> Seq("IHT435_12"),
+    Constants.grossingUpOnEstatePropertyId -> Seq("IHT435_13"),
+    Constants.chargeablePropertyValueId -> Seq("IHT435_14"),
+    Constants.chargeableInheritedPropertyValueId -> Seq("IHT435_15"),
+    Constants.transferAnyUnusedThresholdId -> Seq("IHT435_16"),
+    Constants.valueBeingTransferredId -> Seq("IHT435_17"),
+    Constants.claimDownsizingThresholdId -> Seq("IHT435_18"),
+    Constants.datePropertyWasChangedId -> Seq(
+      "IHT435_20_01",
+      "IHT435_20_02",
+      "IHT435_20_03",
+      "IHT435_20_04",
+      "IHT435_20_05",
+      "IHT435_20_06",
+      "IHT435_20_07",
+      "IHT435_20_08"
+    ),
+    Constants.valueOfChangedPropertyId -> Seq("IHT435_21"),
+    Constants.partOfEstatePassingToDirectDescendantsId -> Seq("IHT435_22"),
+    Constants.grossingUpOnEstateAssetsId -> Seq("IHT435_23"),
+    Constants.valueOfAssetsPassingId -> Seq("IHT435_24"),
+    Constants.transferAvailableWhenPropertyChangedId -> Seq("IHT435_26"),
+    Constants.valueAvailableWhenPropertyChangedId -> Seq("IHT435_27"),
+    Constants.thresholdCalculationResultId -> Seq("IHT435_28")
+  )
+
+  private def setupPDFDocument(pdf:PDDocument) = {
+    val pdDocumentInformation: PDDocumentInformation = pdf.getDocumentInformation
+    pdDocumentInformation.setTitle(Messages("threshold_calculation_result.pdf.title"))
+    pdf.setDocumentInformation(pdDocumentInformation)
+  }
+
+  private def getValueForPDF(jsVal: JsValue, cacheId: String): String = {
+    val dateCacheIds = Set(Constants.dateOfDeathId, Constants.datePropertyWasChangedId)
+    val decimalCacheIds = Set(Constants.percentagePassedToDirectDescendantsId)
+    jsVal match {
+      case n: JsNumber => n.toString
+      case b: JsBoolean => if (b.value) "Yes" else "No"
+      case s: JsString if dateCacheIds.contains(cacheId) =>
+        Transformers.transformDateFormat(s.toString)
+      case s: JsString if decimalCacheIds.contains(cacheId) =>
+        Transformers.stripOffQuotesIfPresent(s.toString).replace(".", "")
+      case s: JsString => s.toString
+      case _ => ""
+    }
+  }
+
+  private def generatePDF(cacheMap: CacheMap) = {
+    val pdf = PDDocument.load(new File("conf/resource/IHT435.pdf"))
+    setupPDFDocument(pdf)
+    val baos = new ByteArrayOutputStream()
+    try {
+      val form = pdf.getDocumentCatalog.getAcroForm
+
+      cacheMapIdToFieldName foreach {
+        case (cacheId, fieldNames) =>
+          val optionalJsVal = cacheMap.data.get(cacheId)
+          optionalJsVal match {
+            case Some(jsVal) =>
+              val valueForPDF = getValueForPDF(jsVal, cacheId)
+              val retrieveValueToStore: (String, Int) => String =
+                if (fieldNames.size == 1) retrieveValueToStoreFor1Field else retrieveValueToStoreForMoreThan1Field
+              var i = 0
```

Comment:
This seems a trifle inelegant - could something be done by doing a foreach on a range perhaps? To eliminat ethe var and the i = i + 1